src/shaders/shaders.js

```diff
@@ -169,6 +178,9 @@ function compile(fragmentSource, vertexSource, isGlobalPrelude) {
         if (preludeFog.staticUniforms) {
             staticUniforms = preludeFog.staticUniforms.concat(staticUniforms);
         }
+        if (preludeShadow.staticUniforms) {
+            staticUniforms = preludeShadow.staticUniforms.concat(staticUniforms);
+        }
```

Comment:
Could we find a way to incur zero cost to both the library size and runtime? While we'd want this to be fully implemented in the library in the future, until it is we don't want to pay the cost for these shader addition. I think the best way would be to skip uniforms scanning, compilation and exclude these new shaders from being bundled.

![Screenshot from 2022-06-14 12-01-09](https://user-images.githubusercontent.com/7061573/173668635-f47b119b-fcdb-4240-867f-8695557cf706.png)

Reply:
Sure, sounds reasonable. I'm not too familiar with the shader compilation and packaging in gl-js. Could you describe how I should do that?

Reply:
@karimnaaji I was able to skip processing the shaders fill_extrusion_depth and ground_shadow. I don't know how to skip _prelude_shadow only when processing shaders for gl-js, but not gl-native. Do you have an idea on how to do that.

Reply:
The only thing I think of right now, is to export a function from shaders.js that takes a parameter whether to include shadow prelude. The function would return what is currently exported from shaders.js. This would need some changes on both gl-js and gl-native sides and I'm not sure if this is the way to go. What do you think?

Reply:
I guess we could also use the approach described in [this article](https://flaming.codes/en/posts/es6-import-with-parameters).

Reply:
>@karimnaaji I was able to skip processing the shaders fill_extrusion_depth and ground_shadow. I don't know how to skip _prelude_shadow only when processing shaders for gl-js, but not gl-native. Do you have an idea on how to do that?

Nice, yes that made the size go down.

>The only thing I think of right now, is to export a function from shaders.js that takes a parameter whether to include shadow prelude. The function would return what is currently exported from shaders.js. This would need some changes on both gl-js and gl-native sides and I'm not sure if this is the way to go. What do you think?

I believe that the `staticUniforms` are unused by native, it's a pre-scanning step that we only use in gl-js. So you could remove that part, and maybe leave it as a note until we support shadows here. Concerning the compilation, maybe export the `compile` function, and then instead of importing these shaders here, you could import them from the `generate-shader` script in native and compile the shader from there instead: https://github.com/mapbox/mapbox-gl-native-internal/blob/internal/scripts/generate-shaders.js#L308, would that work?

Reply:
I managed to skip processing the shadow prelude and now the package size difference is very minor. I think the difference is just the extra code that was added to `fill_extrusion` and `terrain_raster` shaders to receive shadows when the `RENDER_SHADOWS` define is enabled.

![image](https://user-images.githubusercontent.com/4976209/174062617-2e3ef884-8611-4335-b313-84bac7eef2cf.png)

Reply:
Nice! thanks for addressing that :)